app/routes/errors.py

```diff
@@ -92,3 +93,27 @@ def internal_server_error(error=None):
 def http_exception(error):
     log_error(error, error.code)
     return _render_error_page(error.code)
+
+
+@errors_blueprint.app_errorhandler(FulfilmentRequestFailedException)
```

Comment:
This could handle `PublicationFailed` directly and avoid an extra custom exception.